packages/language-server/src/codeActionProvider.ts

```diff
@@ -0,0 +1,189 @@
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import {
+  CodeActionParams,
+  CodeAction,
+  Diagnostic,
+  DiagnosticSeverity,
+  CodeActionKind,
+  Range,
+} from 'vscode-languageserver'
+import { isNullOrUndefined } from 'util'
+import { getAllRelationNames } from './completion/completions'
+import { convertDocumentTextToTrimmedLineArray } from './MessageHandler'
+
+function insertModelOrEnumAtRange(document: TextDocument): Range {
+  // to insert text into a document create a range where start === end.
+  const start = { line: document.lineCount, character: 0 }
+  return { start: start, end: start }
+}
+
+/**
+ * Calculates the Levenshtein Distance
+ * https://en.wikipedia.org/wiki/Levenshtein_distance
+ * adapted from https://leetcode.com/problems/edit-distance/discuss/25846/C%2B%2B-O(n)-space-DP
+ */
+function levenshtein(word1: string, word2: string): number {
```

Comment:
Nice! In Prisma Client we also use levenshtein, I benchmarked the top implementations and found, that `js-levenshtein` was the fastest, but this one should also be fine.

Reply:
Thanks! I tried it out and the `js-levenshtein` is 30% faster, so I'll use that then :)